### ISSUE TITLE
gmtime/localtime bug with day of year

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1959]
Migrated from [rt.perl.org#3232](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3232) (status was 'resolved')

Searchable as RT3232$

**Comment by p5pRT** [comment]
### From schinder@leprss.gsfc.nasa.gov

#### Created by schinder@leprss.gsfc.nasa.gov




leprss% perl
$yday = \(localtime\)\[7\];
print "$yday\\n";
130

leprss% perl
$yday = \(gmtime\)\[7\];
print "$yday\\n";
130

Today is May 10\, the 131st day of the year\.

The pod says&#8203;:

 $yday is the day of the year\, in the range \`1\.\.365' \(or
 \`1\.\.366' in leap years\.\)

Is it missing the fact that this year is a leap year\, or are the docs
wrong?

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.6.0:

Configured by schinder at Thu Mar 23 08:40:31 EST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=solaris, osvers=2.7, archname=sun4-solaris
    uname='sunos leprss.gsfc.nasa.gov 5.7 generic_106541-08 sun4u sparc sunw,ultra-5_10 '
    config_args='-Dcc=gcc -Dprefix=/usr/local -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='gcc', optimize='-O', gccversion=2.95.2 19991024 (release)
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccflags ='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib '
    libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
    libs=-lsocket -lnsl -lgdbm -ldb -ldl -lm -lc -lcrypt -lsec
    libc=/lib/libc.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-fPIC', lddlflags='-G -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /export/home/schinder/Perl/lib
    /export/home/schinder/MacPerl_scripts/lib
    /usr/local/lib/perl5/5.6.0/sun4-solaris
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/sun4-solaris
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl/5.005/sun4-solaris
    /usr/local/lib/perl5/site_perl/5.005
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/export/home/schinder
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=en_US
    LD_LIBRARY_PATH=/usr/local/lib:/lib:/usr/lib:/usr/local/X11R6.4/lib:/usr/openwin/lib:/usr/ucblib
    LOGDIR (unset)
    PATH=.:/export/home/schinder/bin/Perl:/export/home/schinder/bin:/usr/local/X11R6.4/bin:/usr/local/bin:/usr/ucb:/bin:/usr/bin:/usr/openwin/bin:/etc:/usr/etc:/sbin:/usr/sbin:/usr/ccs/bin:/var/qmail/bin:/usr/local/etc:/usr/local/naif/FORTRAN/toolkit/exe:/usr/local/atalk/bin
    PERL5LIB=/export/home/schinder/Perl/lib:/export/home/schinder/MacPerl_scripts/lib
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/tcsh


```
</details>

**Comment by p5pRT** [comment]
### From @tamias

On Wed\, May 10\, 2000 at 09&#8203;:14&#8203;:10PM \-0000\, schinder@&#8203;leprss\.gsfc\.nasa\.gov wrote&#8203;:

> leprss% perl
> $yday = \(localtime\)\[7\];
> print "$yday\\n";
> 130
> 
> leprss% perl
> $yday = \(gmtime\)\[7\];
> print "$yday\\n";
> 130
> 
> Today is May 10\, the 131st day of the year\.
> 
> The pod says&#8203;:
> 
>  $yday is the day of the year\, in the range \`1\.\.365' \(or
>  \`1\.\.366' in leap years\.\)
> 
> Is it missing the fact that this year is a leap year\, or are the docs
> wrong?

This was discussed a bit last month; the docs are wrong on this point; the
range of yday is 0\.\.365\.  I don't know if MJD ever submitted an updated
patch\.

Ronald

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

At 17&#8203;:28 \-0400 2000\-05\-10\, Ronald J Kimball wrote&#8203;:
>This was discussed a bit last month; the docs are wrong on this point; the
>range of yday is 0\.\.365\.  I don't know if MJD ever submitted an updated
>patch\.

Not AFAICT\.  Here's one&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl-5.6.0/pod/perlfunc.pod~        Mon Mar 20 22:13:52 2000
+++ perl-5.6.0/pod/perlfunc.pod Thu May 11 12:24:18 2000
@@ -1983,7 +1983,7 @@
  indicating December.  $year is the number of years since 1900.  That
  is, $year is C<123> in year 2023.  $wday is the day of the week, with
  0 indicating Sunday and 3 indicating Wednesday.  $yday is the day of
-the year, in the range C<1..365> (or C<1..366> in leap years.)
+the year, in the range C<0..364> (or C<0..365> in leap years.)

  Note that the $year element is I<not> simply the last two digits of
  the year.  If you assume it is, then you create non-Y2K-compliant
```
</details>

**Comment by p5pRT** [comment]
### From @tamias

On Thu\, May 11\, 2000 at 12&#8203;:32&#8203;:50PM \+0200\, Dominic Dunlop wrote&#8203;:
> At 17&#8203;:28 \-0400 2000\-05\-10\, Ronald J Kimball wrote&#8203;:
> >This was discussed a bit last month; the docs are wrong on this point; the
> >range of yday is 0\.\.365\.  I don't know if MJD ever submitted an updated
> >patch\.
> 
> Not AFAICT\.  Here's one&#8203;:
> 

I've added a little puncuation fix as well&#8203;:


<details><summary>Inline Patch</summary>

```diff
--- perl-5.6.0/pod/perlfunc.pod~        Mon Mar 20 22:13:52 2000
+++ perl-5.6.0/pod/perlfunc.pod Thu May 11 12:24:18 2000
@@ -1983,7 +1983,7 @@
  indicating December.  $year is the number of years since 1900.  That
  is, $year is C<123> in year 2023.  $wday is the day of the week, with
  0 indicating Sunday and 3 indicating Wednesday.  $yday is the day of
-the year, in the range C<1..365> (or C<1..366> in leap years.)
+the year, in the range C<0..364> (or C<0..365> in leap years).

  Note that the $year element is I<not> simply the last two digits of
  the year.  If you assume it is, then you create non-Y2K-compliant
```
</details>

**Comment by p5pRT** [comment]
perl@tmtm.com - Status changed from 'open' to 'resolved'
